adb_graphics/figures/maps.py

```diff
@@ -287,12 +284,28 @@ def _colorbar(self, cc, ax):
 
         cbar.ax.set_xticklabels(ticks, fontsize=12)
 
-    @utils.timer
-    def draw(self, show=False): # pylint: disable=too-many-locals, too-many-branches
+    def draw(self, show=False):
 
         ''' Main method for creating the plot. Set show=True to display the
         figure from the command line. '''
 
+        cf = self._draw_panel()
+
+        # Draw colorbar
+        self._colorbar(ax=self.map.ax, cc=cf)
+
+        # Finish with the title
+        self._title()
+
+        # Create a pop-up to display the figure, if show=True
+        if show:
+            plt.tight_layout()
+            plt.show()
+
+        self.add_logo(self.map.ax)
+
+    def _draw_panel(self, wind_barbs=True): # pylint: disable=too-many-locals, too-many-branches
```

Comment:
I split out the public `draw()` method here from the `_draw_panel()` method. Now `_draw_panel()` does all the things that are common to single and multi panel plots, and leaves the `draw()` method to do the extras (or not) like plotting colorbars/titles/logos. The addition of a subclass for multipanel plots lets us do a different kind of `draw()` method where we don't do all that by default with each panel.

Reply:
I like this way of doing it.  Nice.